src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -76,17 +76,20 @@ update_shell(unsigned int  index, const std::string& primary, const std::string&
     throw xrt_core::error("Shell not specified");
 
   auto pri = std::make_unique<firmwareImage>(primary.c_str(), MCS_FIRMWARE_PRIMARY);
-  std::unique_ptr<firmwareImage> sec;
   if (pri->fail())
     throw xrt_core::error(boost::str(boost::format("Failed to read %s") % primary));
+
+  std::unique_ptr<firmwareImage> sec;
   if (!secondary.empty()) {
     sec = std::make_unique<firmwareImage>(secondary.c_str(),
       MCS_FIRMWARE_SECONDARY);
     if (sec->fail())
       sec = nullptr;
   }
-
-  flasher.upgradeFirmware("", pri.get(), sec.get());
+  
+  if (flasher.upgradeFirmware("", pri.get(), sec.get()) != 0)
```

Comment:
Minor correction. 
For calls like this it might be best if we simply do the following:
` if (!flasher.upgradeFirmware("", pri.get(), sec.get()))`
Doing so removes the "magic" number and implies we are only interested in failures.  

Unfortunately, the underlying code uses "error codes" and not exceptions so information is lost.  All we know that an error occurred.   For the next release, we should look into update the underlying code to instead:
1. Throw exceptions.
2. Have the exception caught here, reported, and a new exception thrown.

Reply:
I agree!